docs/blog/_posts/2020-02-03-22nd-dotty-milestone-release.md

```diff
@@ -0,0 +1,232 @@
+
+To enable it, you need to run the compiler with the `-Ykind-projector` flag. You can e.g. write the following:
+
+```scala
+// Fix #7139: Implement kind-projector compatibility #7775
+// With -Ykind-projector
+
+trait Functor[F[_]]
+  def map[A, B](fa: F[A], f: A => B): F[B]
+
+object eitherFunctor extends Functor[Either[Int, *]]
+  def map[A, B](fa: Either[Int, A], f: A => B): Either[Int, B] = fa match
+    case Right(x) => Right(f(x))
+    case Left(x) => Left(x)
+
+object functionFunctor extends Functor[Int => *]
+  def map[A, B](fa: Int => A, f: A => B): Int => B =
+    fa andThen f
+
+object tupleFunctor extends Functor[λ[x => (x, x)]]
+  def map[A, B](fa: (A, A), f: A => B): (B, B) = fa match
+    case (a1, a2) => (f(a1), f(a2))
+
+@main def Test =
+  val tpl = (1, 2)
+  val squared = tupleFunctor.map(tpl, a => a * a)
+  println(squared)  // (1,4)
+```
+
+For the discussion, see [PR #7775](https://github.com/lampepfl/dotty/pull/7775). Also see the GitHub [repository](https://github.com/typelevel/kind-projector) of the kind projector Scala 2 plugin for more context.
+
+# Further improvements to the contextual parameters syntax
+Scala 3 contextual parameters are successors of Scala 2 implicits. In Scala 2, they proved useful for a wide range of applications including purely functional programming, dependency injection, type class derivation, type-level programming. Because their apparent value, one of the priorities in Scala 3 for us is to improve the conceptual framework behind them.
+
+The state of contextual parameters before this release heavily employed the `given` keyword. For example:
+
+```scala
+// OLD SYNTAX BELOW
+given String = "10"
+given (given str: String) : Int = str.toInt
+def f(x: Int)(given y: Int) = x * y
+```
+
+The above is a suboptimal solution, however. The feedback we received from the community suggested that many people felt like the `given` keyword was overused, similarly to the `implict` keyword in Scala 2. This overuse is one of the things we'd like to avoid in Scala 3. It leads, for example, to situations like `given (given ...)` which are not nice to read.
+
+For this release, we have changed the syntax for the contextual parameters. The above snippet now becomes:
+
+```scala
+given String = "10"
+given with (str: String) : Int = str.toInt
+def f(x: Int) with (y: Int) = x * y
+```
+
+On the call site, the syntax for explicitly specifying the contextual parameters is now:
+
+```scala
+f(2).with(20)
+```
+
+As opposed to the previous:
+
+```scala
+// OLD SYNTAX BELOW
+f(2)(given 20)
+```
+
+For the time being, the change is experimental and the old syntax is also supported. For the discussion, see [PR #8017](https://github.com/lampepfl/dotty/pull/8017). You can browse the documentation concerning the new syntax [here](https://dotty.epfl.ch/docs/reference/contextual/motivation-new.html).
+
+# Semantics of inline parameters changed
```

Comment:
/cc @nicolasstucki

Reply:
At the end of this section, we should also mention the changes from #8061